### PR TITLE
Deferred API updates

### PR DESCRIPTION
**Pull request by brophdawg11**
Aligns with new changes in https://github.com/remix-run/remix/pull/3434

* Remove `Deferrable`/`ResolvedDeferrable` in favor of raw `Promise`'s and `Awaited`
* Remove generics from `useDeferredData` until `useLoaderData` generic is decided in 6.5
* Support array usages: `return deferred([ await critical() ,lazy() ])`
* Updates `Deferred` `errorElement` to be a legit React error boundary and also handle render errors

### CODE REVIEW

File: packages/react-router/lib/hooks.tsx [hunk 1]
```diff
@@ -724,21 +724,12 @@ export function useRouteError() {
...
-  : T extends Deferrable<infer T2>
-  ? T2 extends Promise<infer T3>
-    ? T3
-    : T2
-  : T;
-
 /**
  * Returns the happy-path data from the nearest ancestor <Deferred /> value
  */
 export function useDeferredData<Data>() {
```
[the remark this answers is not in the document]
Comment (jacob-ebey) [reply packages/react-router/lib/hooks.tsx:730]: I'm honestly not sure here. I'm tempted to say rip it off until we have RR fully integrated in a SSR setup so we don't have any potential breaking changes.

[the remark this answers is not in the document]
Comment (brophdawg11) [reply packages/react-router/lib/hooks.tsx:730]: yeah I'm leaning that way too 🪓

[the remark this answers is not in the document]
Comment (brophdawg11) [reply packages/react-router/lib/hooks.tsx]: https://github.com/remix-run/react-router/pull/9070/commits/4d94cfb10aef8859bbcf2a76fab8e040c417d750

**Comment by brophdawg11** [comment]
support `return deferred(Promise)`

**Pull request merged**